jerry-core/jerry.c

```diff
@@ -2087,3 +2087,46 @@ jerry_dispatch_object_free_callback (ecma_external_pointer_t freecb_p, /**< poin
 
   jerry_make_api_available ();
 } /* jerry_dispatch_object_free_callback */
+
+/**
+ * Reserve a slot in the context for a module
+ *
+ * Returns: 0 or positive on success, -1 on failure.
+ */
+
+/* This value is context-independent */
+static int module_index = -1;
```

Comment:
So, like, slot 0 belongs to `canvas` and slot 1 belongs to `sqlite`, etc. When you start a new context, then slot 0 in that context will belong to the instance of `canvas` loaded into that context, and slot ` in that context will belong to `sqlite` loaded into that context, and the same for every subsequent context.